node/src/command.rs

```diff
@@ -16,15 +16,102 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-use crate::service::Executor;
+use crate::service::MainExecutorDispatch;
 use crate::{
     chain_spec,
     cli::{Cli, Subcommand},
-    service::{self, new_partial},
+    service::{self},
 };
-use primitives::Block;
-use sc_cli::{ChainSpec, Result, Role, RuntimeVersion, SubstrateCli};
-use sc_service::PartialComponents;
+use crate::{
+	cli_relaychain::{RelayChainCli}
+};
+use cumulus_client_service::genesis::generate_genesis_block;
+use parity_scale_codec::Encode;
+use polkadot_parachain::primitives::AccountIdConversion;
+use primitives::{Block, ParaId};
+use sc_cli::{
+    ChainSpec, CliConfiguration, DefaultConfigurationValues, ImportParams, KeystoreParams,
+    NetworkParams, Result, RuntimeVersion, SharedParams, SubstrateCli,
+};
+use sc_service::config::{BasePath, PrometheusConfig};
+use sp_core::hexdisplay::HexDisplay;
+use sp_runtime::traits::Block as BlockT;
+use std::{io::Write, net::SocketAddr};
+
+// default to the Statemint/Statemine/Westmint id
+const DEFAULT_PARA_ID: u32 = 1000;
+
+/// Can be called for a `Configuration` to check what node it belongs to.
+pub trait IdentifyChain {
+    /// Returns if this is a configuration for the `Staking` node.
+    fn is_runtime_staking(&self) -> bool;
+
+    /// Returns if this is a configuration for parachian.
+    fn is_parachain_runtime(&self) -> bool;
+
+    /// Returns if this is a configuration for the `Eden` runtime.
+    fn is_runtime_eden(&self) -> bool;
```

Comment:
No idea is to add prefix `para_xxx` for all parachain diversities, to distinguish from Solo-chain.
first one is used to detect the group of chain like para or solo-chain & second one is used to
detect the chain within para-chain group.
  
```
    fn is_runtime_parachain(&self) -> bool {
        self.id().to_lowercase().starts_with("para")
    }

    fn is_runtime_eden(&self) -> bool {
        self.id().to_lowercase().starts_with("para_eden")
    }
```

Since Service launch procedure for solo & parachin is different, added extra helper function `is_runtime_parachain()`...

![image](https://user-images.githubusercontent.com/82595429/143397941-bfba9ea2-b928-467c-8ad2-c54960c4e308.png)

Reply:
gotcha